File: app/src/main/java/com/elli0tt/rpg_life/presentation/quests/add_quest/AddQuestViewModel.java
package com.elli0tt.rpg_life.presentation.quests.add_quest;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import com.elli0tt.rpg_life.data.repository.QuestsRepositoryImpl;
import com.elli0tt.rpg_life.domain.modal.Quest;

public class AddQuestViewModel extends AndroidViewModel {

    private QuestsRepositoryImpl repository;

    public AddQuestViewModel(@NonNull Application application) {
        super(application);
        repository = new QuestsRepositoryImpl(application);
    }

    public void saveQuest(@NonNull String name,
                          @NonNull String description,
                          @Quest.Difficulty int difficulty) {
        repository.insert(new Quest(name, description, difficulty));
    }


}

File: app/src/main/java/com/elli0tt/rpg_life/presentation/add_quest/AddQuestViewModel.kt
package com.elli0tt.rpg_life.presentation.add_quest

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.elli0tt.rpg_life.data.repository.QuestsRepositoryImpl
import com.elli0tt.rpg_life.domain.model.Quest
import com.elli0tt.rpg_life.domain.use_case.add_edit_quest.InsertQuestsUseCase

class AddQuestViewModel(application: Application) : AndroidViewModel(application) {

    val name = MutableLiveData<String>()

    private val insertQuestByIdUseCase: InsertQuestsUseCase

    init {
        val questsRepository = QuestsRepositoryImpl(application)

        insertQuestByIdUseCase = InsertQuestsUseCase(questsRepository)
    }

    fun saveQuest() {
        val currentName = name.value
        if (currentName != null){
            insertQuestByIdUseCase.invoke(Quest(currentName))
        }

    }


}